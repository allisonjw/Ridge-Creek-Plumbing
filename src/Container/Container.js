import React from 'react';
import './Container.scss';
import Portfolio from '../Portfolio/Portfolio';
import { allProjects } from '../allData/allData';
import { Route } from 'react-router-dom';

export const Container = () => {
    const allProjectsArray = [...allProjects.commercial, ...allProjects.residential, ...allProjects.mixedUse, ...allProjects.hospitality];
    
    return (
        <>
          <Route exact path='/projects/:type' render={({match}) => {
            let projectType = Object.keys(allProjects).find(type => type === match.params.type)
          return <> <Portfolio projectType={allProjects[projectType]} /> </>}}/>  
          <Route exact path='/all' render={() => <> <Portfolio type='all' allProjectsArray={allProjectsArray} /> </>} />    
      </>
    )
};

export default Container;