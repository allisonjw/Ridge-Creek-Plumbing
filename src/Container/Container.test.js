import React from 'react';
import { render } from '@testing-library/react';
import Container from './Container';
import { shallow } from 'enzyme';

describe('Container', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Container />)
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});