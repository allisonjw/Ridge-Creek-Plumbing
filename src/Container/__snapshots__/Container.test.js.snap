// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container should match the snapshot 1`] = `
<Fragment>
  <Route
    exact={true}
    path="/projects/:type"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/all"
    render={[Function]}
  />
</Fragment>
`;
